src/libraries/System.Private.CoreLib/src/System/CharEnumerator.cs

```diff
@@ -31,7 +31,7 @@ public bool MoveNext()
             return false;
         }
 
-        public void Dispose() => _str = null!;
+        public void Dispose() => _str = string.Empty;
```

Comment:
This should have a test added that validates what it is the change is aiming to enable.